lib/matplotlib/font_manager.py

```diff
@@ -855,7 +847,6 @@ def set_weight(self, weight):
         except ValueError:
             if weight not in weight_dict:
                 raise ValueError("weight is invalid")
-            weight = weight_dict[weight]
```

Comment:
It seems that the bug here more that 'hair' is getting mapped to the wrong numeric value.